ui/src/notebooks/shared/Resizer.tsx

```diff
@@ -0,0 +1,175 @@
+// Libraries
+import React, {FC, useRef, useEffect, ReactNode, useState} from 'react'
+import classnames from 'classnames'
+
+// Components
+import ResizerHeader from 'src/notebooks/shared/ResizerHeader'
+
+// Types
+import {IconFont} from '@influxdata/clockface'
+import {PipeData} from 'src/notebooks/index'
+
+// Styles
+import 'src/notebooks/shared/Resizer.scss'
+
+export type Visibility = 'visible' | 'hidden'
+
+interface Props {
+  data: PipeData
+  onUpdate: (data: PipeData) => void
+  children: ReactNode
+  /** If true the resizer can be toggled between Hidden & Visible */
+  toggleVisibilityEnabled: boolean
+  /** If true the resizer cannot be resized, have its visibility toggled, and children will be hidden */
+  resizingEnabled: boolean
+  /** Icon to display in header when resizing is disabled */
+  emptyIcon?: IconFont
+  /** Text to display when resizing is disabled */
+  emptyText: string
+  /** Text to display when the resizer is collapsed */
+  hiddenText?: string
+  /** When resizing is enabled the panel cannot be resized below this amount */
+  minimumHeight?: number
+  /** Renders this element beneath the visibility toggle in the header */
+  additionalControls?: JSX.Element | JSX.Element[]
```

Comment:
the idea being we're gonna add a gear icon for view options? it be better to have a more rigid interface description here if that's the case. like an array of "icon, enabled, onclick, hover text" if it's just all going to be the same buttons, with small differences?